tests/foreman/virtwho/utils.py

```diff
@@ -0,0 +1,185 @@
+            'username': settings.server.ssh_username,
+            'password': settings.server.ssh_password,
+        }
+    else:
+        raise VirtWhoError(
+            '"{}" system type is not supported. Please use one of {}'
+            .format(system_type, ('satellite', 'guest'))
+        )
+
+
+def runcmd(cmd, system=None, timeout=None, port=22):
+    """Return the retcode and stdout.
+    :param str cmd: The command line will be executed in the target system.
+    :param dict system: the system account which ssh will connect to,
+        it will connect to the satellite host if the system is None.
+    :param int timeout: Time to wait for establish the connection.
+    :param int port: The server port to ssh connect tol
+    """
+    system = system or get_system('satellite')
+    hostname = system['hostname']
+    username = system['username']
+    password = system['password']
+    if ":" in hostname:
+        hostname, port = hostname.split(':')
+    result = ssh.command(
+            cmd, hostname=hostname, username=username,
+            password=password, timeout=timeout, port=int(port))
+    return result.return_code, '\n'.join(result.stdout)
+
+
+def register_system(system, activation_key=None, org='Default_Organization'):
+    """Return True if the system is registered to satellite successfully.
+    :param dict system: system account used by ssh to connect and register.
+    :param str activation_key: the activation key will be used to register.
+    :param str org: Which organization will be used to register.
+    :raises: VirtWhoError: If failed to register the system.
+    """
+    runcmd('subscription-manager unregister', system)
+    runcmd('subscription-manager clean', system)
+    runcmd('rpm -qa | grep katello-ca-consumer | xargs rpm -e |sort', system)
+    runcmd(
+        'rpm -ihv http://{}/pub/katello-ca-consumer-latest.noarch.rpm'
+        .format(settings.server.hostname), system
+    )
+    cmd = 'subscription-manager register --org={} '.format(org)
+    if activation_key is not None:
+        cmd += '--activationkey={}'.format(activation_key)
+    else:
+        cmd += '--username={} --password={}'.format(
+            settings.server.admin_username,
+            settings.server.admin_password)
+    ret, stdout = runcmd(cmd, system)
+    if ret == 0 or "system has been registered" in stdout:
+        return True
+    else:
+        raise VirtWhoError(
+            'Failed to register system: {}'.format(system)
+        )
+
+
+def virtwho_cleanup():
+    """Before running test cases, need to clean the environment.
+    Do the following:
+    1. stop virt-who service.
+    2. kill all the virt-who pid
+    3. clean rhsm.log message, make sure there is no old message exist.
+    4. clean all the configure files in /etc/virt-who.d/
+    """
+    runcmd("systemctl stop virt-who")
+    runcmd(
+        "ps -ef | grep virt-who -i | grep -v grep |"
+        "awk '{print $2}' | xargs -I {} kill -9 {}"
+    )
```

Comment:
Because there are two or more virt-who processes exist for the old virt-who package, but now I think we can use pkill to instead, always only one process for the latest virt-who package.